### ISSUE TITLE
`transition_from` test helper adds risk of false positives

### ISSUE THREAD

**Issue by pirj** [aasm/aasm#376]
```
aasm do
  state :a
  state :b
  state :c

  event(:a_to_b) { transitions from: :a, to: :b }
  event(:a_to_c) { transitions from: :a, to: :c }
end

...

# Sane way to assume none of the events allow to transition from :b to :a :
expect(model.new).not_to transition_from(:b).to(:a) # => TypeError: nil is not a symbol nor a string

# Verbose way, number of definitions grow exponentially with number of states and events:
expect(model.new).not_to transition_from(:b).to(:a).on_event(:a_to_b)
expect(model.new).not_to transition_from(:b).to(:a).on_event(:a_to_c)

# Chance to get a false positive by adding a wrong event:
expect(model.new).not_to transition_from(:a).to(:b).on_event(:a_to_c)
# it does transition, but on another event
```

**Comment by taintedzodiac** [comment]
Not entirely sure what the issue is here. Are you saying that the false positive is that you want to test that there are exactly zero transitions allowing the model to go from `:a` to `:b`?

**Comment by weedySeaDragon** [comment]
I echo what taintedzodiac says:  It's not clear what you're trying to point out.

Specifically, re:

    # Sane way to assume none of the events allow to transition from :b to :a :
    expect(model.new).not_to transition_from(:b).to(:a) # => TypeError: nil is not a symbol nor a string


The way to see if you can transition from `:a` to `:b` at all (regardless of the event) is to

    describe do
      let!(:model) { YourModel.new }

      it 'from(:b).to(:a) with any event' do
        model.aasm_state =  :a
        expect(model).not_to allow_event(:b)
     end

**Comment by pirj** [comment]
@weedySeaDragon Compared to `allow_transition_to`, `transition_from` [sets the initial state](https://github.com/aasm/aasm/blob/master/lib/aasm/rspec/transition_from.rb#L4) for you, that's the main difference between those two matchers.

I would really like to be able to use `transition_from` just like `allow_transition_to`, without elaborating on an event (as I do in my example 1).

Because there may be a growing number of events, and you will have to add a test case to make sure none of these events is able to transit from originating state to target state (as I was trying to point out in example two). If you are not careful enough and do not keep your tests up to date when adding a new state, you may get a false positive (e.g. you are missing the case when a newly added transition mistakingly transitions from `dead` to `alive`).

What I propose here is to change the `transition_from` matcher in such a way that it also works if `on_event` is not specified. That may be achieved by changing:

```
    obj.send(@event, *@args) && obj.aasm(@state_machine_name).current_state == @to_state.to_sym
```

to:

```
    if @event
      obj.send(@event, *@args) && obj.aasm(@state_machine_name).current_state == @to_state.to_sym
    else
      # This is taken from allow_transition_to matcher
      obj.aasm(@state_machine_name).states(:permitted => true).include?(state)
    end
```

Adding an additional matcher, say `transitions_between`, that does not take `on_event` in chain is another option.

Bottom line: allow spec'ing out `transition_from` without explicitly specifying `on_event`.

@anilmaurya @taintedzodiac @weedySeaDragon thoughts?

**Comment by weedySeaDragon** [comment]
OK, I see what you're saying.  My definition of 'false positive' is when you run a test and it returns `true` when it really should return `false`.  I think if what you're saying as _omissions_. That's part of what was confusing me.

I'm not convinced this is a good idea.  In a state machine, the only way you can transition from one state to another is via an event.  So what sense does it make to specify a transition  unless you provide an event (via `on_event:`) ?

**Comment by pirj** [comment]
@weedySeaDragon Glad we're on the same page now. Looking over the options I clearly see now that `allow_transition_to` is sufficient, and the original intent to use `expect` on a `model.new` was incorrect in the first place. Sorry it took me so long to figure out that and thanks for patience!
